zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/CuratorXZookeeperDiscoverySpecBuilder.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.zookeeper;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import org.apache.curator.x.discovery.ServiceInstance;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * Builds a {@link ZookeeperDiscoverySpec} for
+ * <a href="https://curator.apache.org/curator-x-discovery/index.html">Curator-X-Discovery</a>.
+ */
+public final class CuratorXZookeeperDiscoverySpecBuilder {
```

Comment:
How about just `CuratorDiscoverySpec(Builder)`? Because:

- X seems to be an acronym for 'extra' or 'extension' (e.g. curator-x-async)
- Curator implies Zookeeper anyway.